### DISCUSSION TITLE
Why does distributed.rs appear to send two messages when replying to a remote actor?

### DISCUSSION THREAD

**Discussion by phil-skillwon** [slawlor/ractor#158] [Q&A] [answered]
On line 116: 
Self::Msg::Rpc(request, reply) => {
	tracing::info!(
		"Received an RPC of '{request}' replying in kind to {} remote actors",
		remote_actors.len()
	);
	let reply_msg = format!("{request}.");
	reply.send(reply_msg.clone())?;
	for act in remote_actors {
		let _reply =
			ractor::call_t!(act, PingPongActorMessage::Rpc, 100, reply_msg.clone())?;
	}
}
at first glance, it seems like a message is sent once: reply.send, 
and then a similar message is sent again to all remote actors: ractor::call_t!(act, PingPongActorMessage::Rpc, 100, reply_msg.clone())?;
I'm a bit confused. Apart from replying in this manner, are there any other methods that can be used as an alternative?

**Comment by slawlor** [comment, marked as answer]
RPC's must have a reply, while non-rpc's don't.

So we first reply, then that actor triggers a broadcast to the other actors waiting on their replies. It's just using RPCs instead of cast operations. It's not really probably the best example, but it's a playground, it's not meant to be production examples.
